### PR TITLE
ExternalContextImpl optimization

### PR DESCRIPTION
**Pull request by pizzi80**
this PR needs the Utils classes of my previous PR, so it's a branch of the other branch

### CODE REVIEW

File: impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java [hunk 1]
```diff
@@ -17,61 +17,32 @@
...
 import com.sun.faces.RIConstants;
 import com.sun.faces.context.flash.ELFlash;
 import com.sun.faces.renderkit.html_basic.ScriptRenderer;
 import com.sun.faces.renderkit.html_basic.StylesheetRenderer;
-import com.sun.faces.util.FacesLogger;
-import com.sun.faces.util.MessageUtils;
-import com.sun.faces.util.TypedCollections;
-import com.sun.faces.util.Util;
-
+import com.sun.faces.util.*;
```
Comment (BalusC) [inline impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java:24]: Can you please reconfigure your IDE to not do that anymore?

File: impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java [hunk 2]
```diff
@@ -130,16 +107,11 @@ public ExternalContextImpl(ServletContext sc, ServletRequest request, ServletRes
 
         boolean enabled = ContextParamUtils.getValue(servletContext, SendPoweredByHeader, Boolean.class);
         if (enabled) {
-            ((HttpServletResponse) response).addHeader("X-Powered-By", "Faces/3.0");
+            ((HttpServletResponse) response).addHeader("X-Powered-By", "Faces/4.0");
```
Comment (BalusC) [inline impl/src/main/java/com/sun/faces/context/ExternalContextImpl.java:110]: Nice catch! This is actually a Mojarra bug which needs to go into 4.0 branch via a separate bugticket.

**Comment by pizzi80** [comment]
fixed imports and further optimized

**Comment by pizzi80** [comment]
@arjantijms  this is just java code optimization ... why not 4.1 ?

which is the Faces 5.0 eta ? december 2023 ?

**Pull request merged**